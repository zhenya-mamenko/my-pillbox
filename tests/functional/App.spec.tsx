import '@testing-library/react-native/extend-expect';
import { render, screen, } from '@testing-library/react-native';
import App from '@/app/index';


jest.mock('@/hooks/useThemeColor', () => (color: string) => {
  const colors: any = {
    background: '#ffffff',
    primary: '#555555',
    surface: '#ffffff',
    text: '#000000',
  }
  return colors[color];
});

jest.mock('@/helpers/localization', () => ({
  __esModule: true,
  default: (s: string) => s
}));

jest.mock('@/components/PillDialog', () => {
  return {
    __esModule: true,
    default: () => <div />,
  };
});

jest.mock('expo-sqlite/kv-store', () => {
  return {
    __esModule: true,
    default: {
      getItemSync: jest.fn(),
      setItem: jest.fn(),
    },
  };
});

jest.mock('expo-font');
jest.mock('expo-asset');

const mockSQLite = {
  deletePill: jest.fn(),
}
jest.mock('@/helpers/sqlite', () => {
  return {
    __esModule: true,
    close: jest.fn(),
    createDatabase: jest.fn(),
    deletePill: (id: number) => mockSQLite.deletePill(id),
    getPills: jest.fn().mockReturnValue([
      {
        id: 1,
        name: 'Pill 1',
        description: 'Description',
        bestBefore: '2025-10-01',
        imageData: 'iVBORw0KGgoAAAANSUhEUgAAAAgAAAAIAQMAAAD+wSzIAAAABlBMVEX///+/v7+jQ3Y5AAAADklEQVQI12P4AIX8EAgALgAD/aNpbtEAAAAASUVORK5CYII',
      },
    ]),
  };
});

jest.mock('react-native-gesture-handler', () => {
  const GS = jest.requireActual('react-native-gesture-handler');
  return {
    __esModule: true,
    ...GS,
    GestureDetector: ({children}: {children: any}) => <div>{children}</div>,
  };
});

jest.mock('react-native-gesture-handler/ReanimatedSwipeable', () => {
  return {
    __esModule: true,
    default: ({children}: {children: any}) => <div>{children}</div>,
  };
});

jest.mock('react-native-reanimated', () => {
  const RNR = jest.requireActual('react-native-reanimated');
  return {
    __esModule: true,
    ...RNR,
    default: {
      ...RNR.default,
      View: ({children}: {children: any}) => <div>{children}</div>,
    }
  }
});

describe('App', () => {

  it('Should render', () => {
    render(<App />);

    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('Should contain right data', () => {
    render(<App />);
    expect(screen.getByText('Pill 1')).toBeTruthy();
    expect(screen.getByText('Description')).toBeTruthy();
    expect(screen.getByText(`pill_best_before: ${(new Date('2025-10-01')).toLocaleDateString()}`)).toBeTruthy();
  });

});
