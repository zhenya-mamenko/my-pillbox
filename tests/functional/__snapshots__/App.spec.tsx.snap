// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={
      [
        {
          "bestBefore": "2025-10-01",
          "description": "Description",
          "id": 1,
          "imageData": "iVBORw0KGgoAAAANSUhEUgAAAAgAAAAIAQMAAAD+wSzIAAAABlBMVEX///+/v7+jQ3Y5AAAADklEQVQI12P4AIX8EAgALgAD/aNpbtEAAAAASUVORK5CYII",
          "name": "Pill 1",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onRefresh={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    refreshing={false}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    style={
      {
        "height": "100%",
        "margin": 0,
        "paddingRight": 10,
        "width": "100%",
      }
    }
    viewabilityConfigCallbackPairs={[]}
  >
    <RCTRefreshControl />
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "flex": 1,
            }
          }
        >
          <div>
            <View
              style={
                {
                  "alignItems": "flex-start",
                  "backgroundColor": "#ffffff",
                  "borderColor": undefined,
                  "borderRadius": 10,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginLeft": 10,
                  "marginRight": 0,
                  "marginTop": 10,
                  "padding": 10,
                }
              }
            >
              <div>
                <div>
                  <ViewManagerAdapter_ExpoImage
                    borderWidth={0.5}
                    containerViewRef={"[React.ref]"}
                    contentFit="cover"
                    contentPosition={
                      {
                        "left": "50%",
                        "top": "50%",
                      }
                    }
                    height="100%"
                    marginBottom={0}
                    marginRight={20}
                    nativeViewRef={"[React.ref]"}
                    onError={[Function]}
                    onLoad={[Function]}
                    onLoadStart={[Function]}
                    onProgress={[Function]}
                    placeholder={[]}
                    source={
                      [
                        {
                          "uri": "data:image/jpg;base64,iVBORw0KGgoAAAANSUhEUgAAAAgAAAAIAQMAAAD+wSzIAAAABlBMVEX///+/v7+jQ3Y5AAAADklEQVQI12P4AIX8EAgALgAD/aNpbtEAAAAASUVORK5CYII",
                        },
                      ]
                    }
                    style={
                      {
                        "borderColor": undefined,
                        "borderWidth": 0.5,
                        "height": "100%",
                        "marginBottom": 0,
                        "marginRight": 20,
                        "width": "100%",
                        "zIndex": 100,
                      }
                    }
                    testID="pill-image-1"
                    transition={null}
                    width="100%"
                    zIndex={100}
                  />
                </div>
              </div>
              <View
                style={
                  {
                    "flex": 1,
                    "flexDirection": "column",
                    "height": "100%",
                    "justifyContent": "space-between",
                    "marginTop": 0,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    {
                      "flexDirection": "column",
                      "verticalAlign": "top",
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#000000",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "marginTop": -5,
                      }
                    }
                    testID="pill-name-1"
                  >
                    Pill 1
                  </Text>
                  <Text
                    ellipsizeMode="tail"
                    numberOfLines={5}
                    style={
                      {
                        "color": "#000000",
                        "fontSize": 14,
                      }
                    }
                    testID="pill-description-1"
                  >
                    Description
                  </Text>
                </View>
                <View>
                  <Text
                    style={
                      {
                        "alignSelf": "flex-end",
                        "color": undefined,
                        "fontSize": 12,
                        "textAlign": "right",
                      }
                    }
                    testID="pill-best-before-1"
                  >
                    pill_best_before: 2025-10-01
                  </Text>
                </View>
              </View>
            </View>
          </div>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "backgroundColor": "transparent",
        "bottom": 0,
        "elevation": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 0,
      }
    }
  >
    <View
      accessibilityLabel="Floating Action Button"
      accessible={true}
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#555555",
          "borderRadius": 28,
          "bottom": 30,
          "elevation": 5,
          "height": 56,
          "justifyContent": "center",
          "opacity": 1,
          "overflow": "visible",
          "position": "absolute",
          "right": 30,
          "shadowColor": "#000000",
          "shadowOffset": {
            "height": 5,
            "width": 0,
          },
          "shadowOpacity": 0.35,
          "shadowRadius": 3,
          "transform": [
            {
              "rotate": "0deg",
            },
            {
              "scale": 1,
            },
          ],
          "width": 56,
          "zIndex": 2,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderRadius": 28,
            "height": 56,
            "justifyContent": "center",
            "opacity": 1,
            "width": 56,
            "zIndex": 3,
          }
        }
      >
        <View
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "borderRadius": 28,
              "flex": 1,
              "height": 56,
              "justifyContent": "center",
              "width": 56,
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/assets/images/add.png",
              }
            }
            style={
              {
                "height": 15,
                "width": 15,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
  <div />
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  />
</View>
`;
